### PR TITLE
Add note about assertions in mocks

### PR DESCRIPTION
**Pull request by Ullaakut**
## Goal of this PR

This PR adds a simple note and includes a code snippet about making assertions in mocks in nominal cases and not in error cases.

### CODE REVIEW

File: content/include/mock_assert_example.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        }
+
+        subject := NewServer()
+
+        got, err := subject.Call()
+        require.NoError(t, err)
+        assert.NotNil(t, got)
+        // Assertions...
+    })
+    t.Run("handles failure to load checkpoint", func(t *testing.T) {
```
Comment (Ullaakut) [inline content/include/mock_assert_example.md:20]:
```suggestion
    t.Run("handles failure to make request", func(t *testing.T) {
```

File: content/include/mock_assert_example.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        t.Parallel()
+
+        mock := mocks.BaselineTransport(t)
+        mock.Request = func(string, string) (*http.Response, error) {
+            return nil, mocks.GenericError()
+        }
+
+        subject := NewServer()
+
+        _, err := tr.BootstrapState(st)
```
Comment (Ullaakut) [inline content/include/mock_assert_example.md:30]:
```suggestion
        subject := NewServer()

        got, err := subject.Call()
```

File: content/include/mock_assert_example.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+    // ...
+    t.Run("nominal case", func(t *testing.T) {
+        t.Parallel()
+
+        mock := mocks.BaselineTransport(t)
```
Comment (emailtovamos) [inline content/include/mock_assert_example.md:5]: I get the overall idea but don't get why the mock isn't used in the example besides its definition.

File: content/include/mock_assert_example.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    })
+    t.Run("handles failure to make request", func(t *testing.T) {
+        t.Parallel()
+
+        mock := mocks.BaselineTransport(t)
+        mock.Request = func(string, string) (*http.Response, error) {
+            return nil, mocks.GenericError()
+        }
+
+        subject := NewServer()
```
Comment (Ullaakut) [inline content/include/mock_assert_example.md:28]:
```suggestion
        subject := NewServer(mock)
```

File: content/include/mock_assert_example.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+    // ...
+    t.Run("nominal case", func(t *testing.T) {
+        t.Parallel()
+
+        mock := mocks.BaselineTransport(t)
+        mock.Request = func(method, url string) (*http.Response, error) {
+            assert.Equal(t, http.MethodPost, method)
+            assert.Equal(t, "https://google.com", url)
```
Comment (ricardomgoncalves) [inline content/include/mock_assert_example.md:8]: Should this be require? It should stop here right?

Comment (Ullaakut) [reply content/include/mock_assert_example.md:8]: No, I don't see why this should be a `require` 🤔 We're just checking that we called the mock with the right parameters. The only case where we want to use require is when we make an assertion which if it fails, will for sure cause others to fail or cause panics down the line. For example if we're making assertions on `x.AttributeY` and `x.AttributeZ` we probably want to first do `require.NotNil(t, x)`. If we're making assertions on something returned along with an error, we probably want to do `require.NoError(t, err)` before making assertions on that thing. If we're making setup calls we want to require for them not to error.

Does that make sense? Maybe I should add another note to clarify when to use assert and when require then 😄

**Review by ricardomgoncalves — APPROVED** [review]
LGTM

File: content/include/mock_assert_example.md (hunk 5 above)
Comment (ricardomgoncalves) [reply content/include/mock_assert_example.md:8]: No need, I thought you wanted to fail the test in the spot when some of the arguments don't match. I think that is how gomock works.

**Review by emailtovamos — APPROVED** [review]

**Pull request merged**